code/modules/mining/ores_coins.dm

```diff
@@ -125,7 +126,7 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
 	AddComponent(/datum/component/storm_hating)
 
 /obj/item/stack/ore/glass/on_orm_collection() //we need to smelt the glass beforehand because the silo and orm don't accept sand mats
-	var/obj/item/stack/sheet/glass = new refined_type(drop_location(), amount)
+	var/obj/item/stack/sheet/glass = new refined_type(null, amount) //spawned in null loc, otherwise the newly spawned glass would be collected separately, causing issues.
```

Comment:
```suggestion
	var/obj/item/stack/sheet/glass = new refined_type(drop_location(), amount, merge = FALSE) //The newly spawned glass should not merge with other stacks on the turf, else it could cause issues.
```

Your approach works as well. Nobody likes null space though so it's upto you. 

Also fun fact this problem never occurs because it is impossible for 2 piles of sand that is less than 50 stacks to exist in the same location because they would just merge.
And when you do have say 2 piles of sand that is 50 stacks each the glass it creates will not merge because 50 is the merge limit for all stacks
Still good to check though